engine/buildkit/telemetry.go

```diff
@@ -2,58 +2,83 @@ package buildkit
 
 import (
 	"context"
+	"log/slog"
 	"strings"
 
+	"github.com/moby/buildkit/client/llb"
+	"github.com/opencontainers/go-digest"
 	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
 	sdklog "go.opentelemetry.io/otel/sdk/log"
 	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
 	"go.opentelemetry.io/otel/trace/embedded"
+	"go.opentelemetry.io/otel/trace/noop"
 
 	"dagger.io/dagger/telemetry"
 )
 
+func WithTracePropagation(ctx context.Context) llb.ConstraintsOpt {
+	mc := propagation.MapCarrier{}
+	telemetry.Propagator.Inject(ctx, mc)
+	return llb.WithDescription(mc)
+}
```

Comment:
Sorry, struggling to understand what the effect of calling this actually is :sweat_smile: